app/controllers/history.js

```diff
@@ -2,8 +2,10 @@ import Ember from 'ember';
 
 export default Ember.Controller.extend({
 
-  sortProperties: ['name:desc'],
-  sortedModel: Ember.computed.sort('model', 'sortProperties'),
+  queryChanged: function() {
+    this.set('filteredModel', this.get('model').filter( item => item.name.toLowerCase().includes(this.get('q'))).sort((a, b) => a <= b ? 1 : -1));
+  }.observes('q', 'model'),
```

Comment:
In a lot of cases, using an observer is considered bad practice.
Here, you can just define `filteredModel` to be a computed property that depends on the keys `q` and `model` instead.